flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/aggregations.scala

```diff
@@ -52,12 +62,49 @@ abstract sealed class Aggregation extends Expression {
 
 }
 
+case class DistinctAgg(child: Expression) extends Aggregation {
+
+  override private[flink] def toAggCall(name: String)(implicit relBuilder: RelBuilder): AggCall = {
+    child.asInstanceOf[Aggregation].toAggCall(name, isDistinct = true)
+  }
+
+  override private[flink] def resultType: TypeInformation[_] = child.resultType
+
+  override private[flink] def validateInput(): ValidationResult = {
+    super.validateInput()
+    child match {
+      case agg: Aggregation =>
+        child.validateInput()
+      case _ =>
+        ValidationFailure(s"Distinct modifier cannot be applied to $child. " +
+          s"It can only be applied to an aggregation expression!")
```

Comment:
It can only be applied to an aggregation expression, for example, 'a.count.distinct which means COUNT(DISTINCT a).
Add an example may be more understandable?